consensus/spos/bls/v2/subroundEndRound.go

```diff
@@ -260,9 +261,18 @@ func (sr *subroundEndRound) doEndRoundJobByNode() bool {
 		if !sr.waitForSignalSync() {
 			return false
 		}
-		sr.sendProof()
-		err := sr.prepareBroadcastBlockData()
```

Comment:
previously whatever happened in the `sendProof` the `sr.prepareBroadcastBlockData()` was always called. In the new version, if anything failed in the sendProof the `sr.prepareBroadcastBlockData()` the first bool returned is false and the method not called. This is an intended optimization?

Reply:
ok